src/main/java/net/minecraftforge/event/entity/player/DamageBonusEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+
+/**
+ * This event is fired whenever a player attacks an Entity in
+ * Player#attack(Entity).<br>
+ * <br>
+ * This event is not {@link Cancelable}.<br>
+ * <br>
+ * This allows adding to the base damage like DamageEnchantments do, but also considering the attacked Entity object including capabilities, not just the Entity's MobType
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+ **/
+public class DamageBonusEvent extends PlayerEvent
+{
+    private final ItemStack weapon;
+    private final Entity target;
+    private float bonus;
+
+    public DamageBonusEvent(Player player, ItemStack weapon, Entity target, float enchantmentBonus)
+    {
+        super(player);
+        this.weapon = weapon;
+        this.target = target;
+        this.bonus = enchantmentBonus;
+    }
+
+    /**
+     * @return The attacked entity
+     */
+    public Entity getTarget()
+    {
+        return target;
+    }
+
+    /**
+     * @return The item used for the attack
+     */
+    public ItemStack getWeapon()
+    {
+        return weapon;
+    }
+
+    /**
+     * @param additionalDamage Added to the existing bonus damage
+     */
+    public void addBonus(float additionalDamage)
+    {
+        this.bonus += additionalDamage;
+    }
```

Comment:
I would probably recommend making this a `setBonus` method instead of add as most Forge methods uses the `set` syntax so reclaculation may be different. This could be an additional method, but there should be something to just set raw.